spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -517,20 +566,140 @@ function chanCloseConfirm(
     connection = provableStore.get(connectionPath(channel.connectionHops[0]))
     abortTransactionUnless(connection !== null)
     abortTransactionUnless(connection.state === OPEN)
+
+    let counterpartyHops: []string = []
+    if (connectionHops.length > 1) {
+      counterpartyHops = getCounterPartyHops(proofInit, connection)
+    } else {
+      counterpartyHops = [connection.counterpartyConnectionIdentifier]
+    }
+
     expected = ChannelEnd{CLOSED, channel.order, portIdentifier,
-                          channelIdentifier, [connection.counterpartyConnectionIdentifier], channel.version}
-    abortTransactionUnless(connection.verifyChannelState(
-      proofHeight,
-      proofInit,
-      channel.counterpartyPortIdentifier,
-      channel.counterpartyChannelIdentifier,
-      expected
-    ))
+                          channelIdentifier, counterpartyHops, channel.version}
+
+    if (connectionHops.length > 1) {
+      key = host.ChannelPath(counterparty.PortId, counterparty.ChannelId)
+
+      abortTransactionUnless(connection.verifyMultihopMembership(
+        connection,
+        proofHeight,
+        proofInit,
+        channel.connectionHops,
+        key
+        expected))
+    } else {
+      abortTransactionUnless(connection.verifyChannelState(
+        proofHeight,
+        proofInit,
+        channel.counterpartyPortIdentifier,
+        channel.counterpartyChannelIdentifier,
+        expected
+      ))
+    }
+
     channel.state = CLOSED
     provableStore.set(channelPath(portIdentifier, channelIdentifier), channel)
 }
 ```
 
+The `chanCloseFrozen` function is called by a relayer to force close a multi-hop channel if any client state in the
+channel path is frozen. A relayer should send proof of the frozen client state to each end of the channel with a
+proof of the frozen client state in the channel path starting from each channel end up until the first frozen client.
+The multi-hop proof for each channel end will be different and consist of a proof formed starting from each channel
+end up to the frozen client.
+
+The multi-hop proof starts with a chain with a frozen client for the misbehaving chain. However, the frozen client exists
+on the next blockchain in the channel path so the key/value proof is indexed to evaluate on the consensus state holding
+that client state. The client state path requires knowledge of the client id which can be determined from the
+connectionEnd on the misbehaving chain prior to the misbehavior submission.
+
+Once frozen, it is possible for a channel to be unfrozen (reactivated) via governance processes once the misbehavior in
+the channel path has been resolved. However, this process is out-of-protocol.
+
+```typescript
+function chanCloseFrozen(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  proofConnection: MultihopProof,
+  proofClientState: MultihopProof,
```

Comment:
There is a multihop proof of the frozen client and a multihop proof of the associated connection. Both are required for the proof and multihop is required if the frozen client is midway through the channel.

It might be nice to be able to combine the proofs here if that became supported.